android/tangram/src/com/mapzen/tangram/FontFileParser.java

```diff
@@ -20,6 +21,8 @@
 
     private void processDocument(XmlPullParser parser) throws XmlPullParserException, IOException {
 
+        Vector<String> weights = new Vector<String>();
```

Comment:
This list should be cleared whenever a new "family" is encountered. 

I'd also recommend `ArrayList` here instead of `Vector`; the difference between the two is that `Vector` synchronizes for thread safety on every operation, making it generally slower than `ArrayList`